home-work-objects-in-details.js

```diff
@@ -0,0 +1,76 @@
+// 1. Задача про обчислення різниці часу
+
+function durationBetweenDates(startDate = '24 Aug 1991', endDate = '24 Aug 1991', dimension = 'days') {
+  const start = new Date(startDate);
+  const end = new Date(endDate);
+
+  const timeDiff = Math.abs(end - start); // time difference in milliseconds
+
+  switch (dimension) {
+    case 'days':
+      const daysDiff = Math.floor(timeDiff / (1000 * 60 * 60 * 24));
+      return `${daysDiff} days`;
+    case 'hours':
+      const hoursDiff = Math.floor(timeDiff / (1000 * 60 * 60));
+      return `${hoursDiff} hours`;
+    case 'minutes':
+      const minutesDiff = Math.floor(timeDiff / (1000 * 60));
+      return `${minutesDiff} minutes`;
+    case 'seconds':
+      const secondsDiff = Math.floor(timeDiff / 1000);
+      return `${secondsDiff} seconds`;
+  }
```

Comment:
There is a lot of copy/paste. Do you see it?
To avoid it let’s create an object with keys - dimensions, and value - numbers.